File: project-euler.java/src/main/java/euler/solutions/_16.java
package euler.solutions;

import euler.common.BaseSolution;

import java.math.BigInteger;

public class _16 extends BaseSolution {

    @Override
    public long solve() {
        String twoPow1000 = new BigInteger("2").pow(1000).toString();
        int sum = 0;
        for (int i = 0; i < twoPow1000.length(); i++) {
            sum += twoPow1000.charAt(i) - 48;
        }
        return sum;
    }
}

File: project-euler.go/solutions/_16/16.go
package _16

import "math/big"

func Solve() int64 {
	var twoPow1000 big.Int
	twoPow1000.Exp(big.NewInt(2), big.NewInt(1000), nil)

	twoPow1000Str := twoPow1000.String()

	var sum int64
	for i := range twoPow1000Str {
		sum += int64(twoPow1000Str[i] - 48)
	}

	return sum
}
